File: debrujin/ruby/logic/1.0.0.rb
def debrujin(alpha, n)
  string = alpha[0] * n
  alphaLocation = 0
  while string.length < base=alpha.length ** n - 1 + n
    if string[string[1 - n, string.length] + beta = alpha[alphaLocation]]
      alphaLocation += 1
      if alphaLocation >= base
        alphaLocation = alpha.index(string[-1]) + 1
        string = string[0...-1]
      end
    else
      string += beta
      alphaLocation = 0
    end
  end
  return string
end

File: debrujin/python/logic/1.0.0.py
def debrujin(alpha, n):
  base = len(alpha)
  string = alpha[0] * n
  alphaLocation = 0
  while len(string) < int(str(base - 1) * n, base) + n:
    beta = alpha[alphaLocation]    
    if string.find(string[1 - n:] + beta) + 1:
      alphaLocation += 1
      if alphaLocation >= base:
        alphaLocation = alpha.find(string[-1:]) + 1
        string = string[:-1]
    else:
      string += beta
      alphaLocation = 0
  return string